File: src/modifiers/toolbarTitleMenu.ts
import type { ReactNode } from 'react';
import { createModifier } from './createModifier';

export interface ToolbarTitleMenuParams {
  /// The menu's content — typically `Button`s — shown when the user taps the
  /// navigation title.
  content: ReactNode;
}

/// SwiftUI `.toolbarTitleMenu { … }` (watchOS 9+). Populates the menu that
/// appears from the navigation title.
export function toolbarTitleMenu(content: ReactNode): ReturnType<typeof createModifier>;
export function toolbarTitleMenu(
  params: ToolbarTitleMenuParams
): ReturnType<typeof createModifier>;
export function toolbarTitleMenu(a: ToolbarTitleMenuParams | ReactNode) {
  if (a && typeof a === 'object' && 'content' in a) {
    return createModifier('toolbarTitleMenu', a as ToolbarTitleMenuParams);
  }
  return createModifier('toolbarTitleMenu', { content: a as ReactNode });
}

File: apple/Sources/ReactNativeWatchOS/Modifiers/ToolbarTitleMenu.swift
import SwiftUI

/// SwiftUI `.toolbarTitleMenu { … }` (watchOS 9+). Populates the menu shown
/// from the navigation title. The `content` slot is resolved via `ctx`; with
/// no content provided the view is returned unchanged.
enum RNWToolbarTitleMenuModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("toolbarTitleMenu") { view, params, ctx in
            guard let body = ctx.content(params.string("content")) else {
                return view
            }
            return AnyView(view.toolbarTitleMenu { body })
        }
    }
}
